systests/cdi/base/src/main/java/org/apache/cxf/systests/cdi/base/BookStoreVersion.java

```diff
@@ -22,20 +22,25 @@
 import javax.enterprise.context.RequestScoped;
 import javax.inject.Inject;
 import javax.ws.rs.GET;
-import javax.ws.rs.core.Context;
 import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
+import org.apache.cxf.systests.cdi.base.context.CustomContext;
+
 @RequestScoped
 public class BookStoreVersion {
     @Inject
     private String version;
-    @Context
+    @Inject
     private HttpHeaders httpHeaders;
+    @Inject
```

Comment:
@sberyozkin @reta  I'm a bit stumped at this point.  Even when I annotate the field with `@Context`, I don't see it being processed using my `ContextProvider` yet somehow there is a value set..

Reply:
@johnament Not sure I understood the exact nature of the problem, but with `@Context` annotation the `CustomContext` is being processed by `DelegateContextAnnotatedType` which injects into it the proxy value (CDI managed one), and later on the instance is created by `ContextProducerBean` on first invocation.  Is it something along this lines or absolutely different flow?

Reply:
@reta yes, that's what I'm expecting to happen, however I'm never seeing the bean invoked for this context object.

Reply:
@johnament I conducted a short debug session, it looks like it is working as expected / described, I see `DelegateContextAnnotatedType`, `ContextProducerBean` and finally `CustomContextImpl` being invoked in case when `CustomContext` is annotated with `@Context` annotation. Interesting, what is happening in your case ...